drivers/linstorvolumemanager.py

```diff
@@ -1439,6 +1439,24 @@ def find_up_to_date_diskful_nodes(self, volume_uuid):
 
         return (node_names, in_use_by)
 
+    def get_primary(self, volume_uuid):
+        """
+        Find the node that opened a volume, i.e. the primary.
+        :rtype: str
+        """
+        volume_name = self.get_volume_name(volume_uuid)
+
+        resource_states = filter(
+            lambda resource_state: resource_state.name == volume_name,
+            self._get_resource_cache().resource_states
+        )
+
+        for resource_state in resource_states:
+            if resource_state.in_use:
+                return resource_state.node_name
+
+        return None
```

Comment:
Not used? (I know you are not the author here. :smile: )

Reply:
Should I remove it or let the original author argue / deal with it ?

Reply:
I removed the function and squashed all of my commits together